cassandra-mesos-model/src/main/proto/io/mesosphere/mesos/frameworks/cassandra/model.proto

```diff
@@ -742,9 +748,13 @@ message NodeJobTask {
      */
     required ClusterJobType jobType = 1;
     /**
-     * Backup name
+     * Backup directory
+     */
+    optional string backupDir = 3;
+    /**
+     * Determine if this is the first node participating in job
      */
-    optional string backupName = 2;
+    optional bool firstNodeInJob = 4;
```

Comment:
Hi @elingg,
I've extracted TRUNCATE job and updated logic of NodeTaskClusterJobHandler instead of adding RestoreClusterJobHandler (99% duplicated code). 

Now the Scheduler decides what jobs run where, as you and Ben wanted.
I am going to update the docs now. Please merge after that.